contributed/ggStrongLensingDC2_SL_NanLi_HackurDC2.ipynb

```diff
@@ -0,0 +1,752 @@
+    "    return res\n",
+    "\n",
+    "def Dc2(z1,z2):\n",
+    "    Dcz1 = (cosmo.comoving_distance(z1).value*cosmo.h)\n",
+    "    Dcz2 = (cosmo.comoving_distance(z2).value*cosmo.h)\n",
+    "    res = Dcz2-Dcz1+1e-8\n",
+    "    return res\n",
+    " \n",
+    "def Da(z):\n",
+    "    res = cosmo.comoving_distance(z).value*cosmo.h/(1+z)\n",
+    "    return res\n",
+    " \n",
+    "def Da2(z1,z2):\n",
+    "    Dcz1 = (cosmo.comoving_distance(z1).value*cosmo.h)\n",
+    "    Dcz2 = (cosmo.comoving_distance(z2).value*cosmo.h)\n",
+    "    res = (Dcz2-Dcz1+1e-8)/(1.0+z2)\n",
+    "    return res\n",
+    "\n",
+    "def Dl(z): \n",
+    "    res = cosmo.luminosity_distance(z).value*cosmo.h \n",
+    "    return res \n",
+    "\n",
+    "def SigmaCrit(z1,z2):\n",
+    "    res = (vc*vc/4.0/np.pi/G*Dc(z2)/(Dc(z1)/(1.0+z1))/Dc2(z1,z2))\n",
+    "    return res"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def mags_to_vd(mg,mr,zz):\n",
+    "    '''\n",
+    "        Faber et al. 2007 \n",
+    "        Parker et al. 2007, Table 1\n",
+    "        Need more discussion on involving more realistic models.\n",
+    "    '''\n",
+    "    Dlum     = Dl(zz)\n",
+    "    Mabsr    = mr-5.0*np.log10(Dlum/cosmo.h)-25.0\n",
+    "    mrsdss   = Mabsr+0.024*(mg-mr)/0.871\n",
+    "    mrsdss   = mrsdss-0.11\n",
+    "    mrstar   = (-20.44)+(zz-0.1)*1.5\n",
+    "    LbyLstar = 10.0**(-0.4*(mrsdss-mrstar))\n",
+    "    res      = 142.0*LbyLstar**(1./3.)\n",
+    "    return res\n",
+    "\n",
+    "def re_sv(sv,z1,z2):\n",
+    "    res = 4.0*np.pi*(sv**2.0/vc**2.0)*Da2(z1,z2)/Da(z2)*apr\n",
+    "    return res\n",
+    "\n",
+    "def sersic_2d_hist_phosim(xi1, xi2, xc1, xc2, a_major, b_minor, pha, ndex=1.0):\n",
+    "    bn = 2.0*ndex-1/3.0+0.009876/ndex\n",
+    "    phirad = np.deg2rad(pha)\n",
+    "    xi1new = (xi1-xc1)*np.cos(phirad)+(xi2-xc2)*np.sin(phirad)\n",
+    "    xi2new = (xi2-xc2)*np.cos(phirad)-(xi1-xc1)*np.sin(phirad)\n",
+    "    R_scale = np.sqrt((xi1new/a_major)**2+(xi2new/b_minor)**2)\n",
+    "    R_scale[R_scale<0.01] = 0.01\n",
+    "    img = np.exp(-bn*((R_scale)**(1.0/ndex)-1.0))\n",
+    "    res = img/img.sum()\n",
+    "    return res\n",
+    "\n",
+    "def ftot_mb(mag_in, zpoint):\n",
+    "    res = 10.0**((zpoint - mag_in)*0.4)\n",
+    "    return res\n",
+    "\n",
+    "def make_c_coor(bs,nc):\n",
+    "    ds = bs/nc\n",
+    "    xx01 = np.linspace(-bs/2.0,bs/2.0-ds,nc)+0.5*ds\n",
+    "    xx02 = np.linspace(-bs/2.0,bs/2.0-ds,nc)+0.5*ds\n",
+    "    xi2,xi1 = np.meshgrid(xx01,xx02)\n",
+    "    return xi1,xi2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def sie_alphas_simple(x0, y0, theta, ql, re, le, ext_shears, ext_angle, ext_kappa, x, y, rc=0.0):  # SIE lens model\n",
+    "    \n",
```

Comment:
Could you give a little more comment here, @linan7788626 (e.g. citation), so that new users can follow more closely?

Reply:
more descriptions and references have been added, please feel free to let me know if any further actions are needed.